### ISSUE TITLE
Obsolete Azure CLI commands and YAML specs in MS Learn

### ISSUE THREAD

**Issue by carlosjuribe** [MicrosoftLearning/mslearn-azure-ml#86]
# Training Module: Train models in Azure Machine Learning with the CLI (v2)
### Section: Create a dataset asset

> Hi, I'm creating this issue here because the [original page in MS Learn](https://learn.microsoft.com/en-us/training/modules/create-azure-machine-learning-resources-cli-v2/4-manage-workspace-assets) does not allow me to submit feedback more than once. As I already submitted feedback on one typo, here it goes the new feedback.

Versions:
- `azure-cli`: 2.59.0
- extension `ml`: 2.25.1

### 1. What appears an obsolete command

![image](https://github.com/MicrosoftLearning/mslearn-azure-ml/assets/46844109/89b48275-7947-47fb-adda-377dc9b2a6ab)

The command `az ml dataset list` throws this error: `'dataset' is misspelled or not recognized by the system.
Did you mean 'datastore' ?`. It looks like the command is obsolete now, as to work it should say **data**, not **dataset**, like `az ml data list`. Same goes for the command above it, `az ml dataset create --file data-local-path.yml`.

### 2. What appears an obsolete YAML file for data asset properties

In [the same page](https://learn.microsoft.com/en-us/training/modules/create-azure-machine-learning-resources-cli-v2/4-manage-workspace-assets) as in the point above, the YAML that specifies the dataset asset properties didn't work due to two typos:

![image](https://github.com/MicrosoftLearning/mslearn-azure-ml/assets/46844109/153fab93-fc29-4195-ba5e-630ad11130d9)

- (yellow in image) **Wrong parameter name in the YAML**
    When I ran the command `az ml data create --file data-local-path.yml` I got the error: 
```
(x) path:
- Missing data for required field.

(x) local_path:
- Unknown field.
```
So looks like `path` replaced `local_path`. 

- (blue in image) **Wrong value for the `path` parameter**: Having saved a CSV named `"customer-churn.csv"` at the same level that `data-local-path.yml`, when running `az ml data create --file data-local-path.yml` I got the error:
```
(x) File path does not match asset type uri_folder: /mnt/batch/.../customer-churn.csv
``` 
Thus it seems it expects a URI folder and not a path to the final data file. I fixed that by putting the `customer-churn.csv` file in a folder `"datasets"` and then specifying just this folder inside the `data-local-path.yml` YAML (_i.e._, `path: datasets`). Then a re-run of the `az ml data create` command worked fine:

![image](https://github.com/MicrosoftLearning/mslearn-azure-ml/assets/46844109/bc6461d6-47da-4b4b-967b-95e0548914d1)

### Conclusion

In summary, the YAML specs and the command that work are:

The local data file inside a folder
```bash
datasets/
└── customer-churn.csv
```
The YAML file with the **path** field and the value specifying the path to a folder (not a file):
```yaml
# data-local-path.yml
$schema: https://azuremlschemas.azureedge.net/latest/asset.schema.json
name: customer-churn-data
version: 1
path: datasets
description: Dataset pointing to customer churn CSV on local computer. Data will be uploaded to default datastore
```
The corrected Azure CLI command with **data** instead of **dataset**:

`az ml data create --file aml_data_asset.yml`

**Comment by carlosjuribe** [comment]
Sorry, closing this issue as this is not the proper repo for it. My bad. I re-created it at [the correct repo](https://github.com/MicrosoftLearning/mslearn-aml-cli/issues/13).
